src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -286,10 +286,30 @@
           (app (assoc req :body-string payload))
           (http/error-response (str "Missing required parameter 'payload'")))
         "application/json"
-        (let [body-string (request/body-string req)]
-          (if (nil? body-string)
+        (let [{:strs [certname version command]} params
```

Comment:
If I'm not misunderstanding, we already have a command-app handler that ensures the incoming request will be application/jjson, so I think we might be able to considerably simplify this function and the related test function.